packages/opentelemetry-resources/src/types.ts

```diff
@@ -26,9 +26,10 @@ import { SpanAttributes } from '@opentelemetry/api';
 export type ResourceAttributes = SpanAttributes;
 
 /**
- * Interface for a Resource Detector. In order to detect resources in parallel
- * a detector returns a Promise containing a Resource.
+ * Interface for a Resource Detector. In order to detect resources asynchronously, a detector
+ * can pass a Promise as the second parameter to the Resource constructor. Returning a
+ * Promise<Resource> is deprecated in favor of this approach.
  */
 export interface Detector {
-  detect(config?: ResourceDetectionConfig): Promise<Resource>;
+  detect(config?: ResourceDetectionConfig): Promise<Resource> | Resource;
```

Comment:
@open-telemetry/javascript-maintainers Can this be considered a breaking change? Users _shouldn't_ be calling resource detectors directly but if they are I can easily see some code like `detect().then(something)` breaking (TypeError: detect(...).then is not a function).

Reply:
I think this is a breaking change.

Reply:
#3055